### DISCUSSION TITLE
Implement Wi-Fi management and Cluster Handlers for Wi-Fi network

### DISCUSSION THREAD

**Discussion by Apollon77** [matter-js/matter.js#256] [Feature requests (vote via thumbs up)]
In order to allow e.g. "Linux based non-Ethernet Devices" we would need an Implementation of a Wi-Fi network Commissioning cluster with all relevant handlers to scan, set and connect to Wi-Fi networks (could be done via wpa-supplicant for Linux based platforms) which uses BLE as commissioning path. With this "Raspi like" platforms without ethernet (raspberry Zero W like) could also act as a devices when BLE and Wi-Fi are supported.

A pottential (second step!) enhancement could be the support of Soft-AP (open a Wi-Fi hotspot for initial commissioning connection) as commissioning option when VLE is not available. This could be done on linux using hostapd. Would potentially require W-Fi control options (see above) for Controller side too.

**Comment by adminy** [comment]
Just to clarify from my understanding of our discussion earlier, This is basically a way for devices to gain wifi credentials in the initial setup over BLE / QR. So that the device is able to communicate via Wifi once the initial setup has completed. So this is basically going to expand on the setup part to do more based on the device's capabilities and connect automatically.

**Reply by Apollon77** [reply]
yes
